### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by jinapark202**
Added info from internal doc to Readme with table of contents.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,6 +139,383 @@ Here's a full example:
...
+    | 'regex' // tests whether a string is a valid regular expression by passing it to RegExp constructor.
+    | 'uuid' // Universally Unique IDentifier according to RFC4122.
+    | 'password' // hint to the UI to hide/obfuscate input strings (applied automatically when using `type: 'password'`
+    | 'text' // longer strings (applied automatically when using `type: 'text'`
+}
+```
+
+## Dynamic Dropdowns
+
+Some APIs require users to specify a related object or resource by id. Unfortunately, this is rather unintuitive for people who don’t speak or memorize ids. Dynamic dropdowns offer users a way to select those ids with human-readable labels.
```
Comment (mericsson) [inline README.md:280]: This is not functional yet. It is on our roadmap. We can either remove it or make suggested change.

```suggestion
## Dynamic Dropdowns

_Not yet available, please contact Segment team if this is something you need._

Some APIs require users to specify a related object or resource by id. Unfortunately, this is rather unintuitive for people who don’t speak or memorize ids. Dynamic dropdowns offer users a way to select those ids with human-readable labels.
```

File: docs/authentication.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  extendRequest({ auth }) {
+    return {
+      headers: {
+        authorization: `Bearer ${auth?.accessToken}`
+      }
+    }
+  }
+```
+
+**\*Note:** OAuth directly depends on the oauth providers available in oauth-service. You have to follow the process of Adding a new OAuth provider before using the oauth2 scheme in an action.\*
```
Comment (mericsson) [inline docs/authentication.md:149]:
```suggestion
**\*Note:** OAuth directly depends on the oauth providers available in Segment's internal OAuth Service. Please contact Segment if you require OAuth for your destination.\*
```

File: docs/authentication.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      headers: { Authorization: `Bearer ${settings.apiKey}` },
+      responseType: 'json'
+    }
+  }
+}
+```
+
+## OAuth2 Authentication Scheme
+
+OAuth2 Authentication scheme is the model to be used for destination APIs which support [OAuth 2.0](https://oauth.net/2/). You’ll be able to define a `refreshAccessToken` function if you want the framework to refresh expired tokens.
```
Comment (mericsson) [inline docs/authentication.md:103]:
```suggestion
## OAuth2 Authentication Scheme

_oauth authentication is not generally available to external partners as part of Developer Center Pilot. Please contact Segment team if you require it._

OAuth2 Authentication scheme is the model to be used for destination APIs which support [OAuth 2.0](https://oauth.net/2/). You’ll be able to define a `refreshAccessToken` function if you want the framework to refresh expired tokens.
```

**Comment by mericsson** [comment]
Thanks! Left a few comments, otherwise lgtm! Thanks!

File: README.md [hunk 4]
```diff
@@ -125,6 +138,335 @@ Here's a full example:
...
+
+Keep in mind a few important things about how batching works:
+
+- Batching can add latency while Segment accumulates events in batches internally. This can be up to a minute, currently, but this is subject to change at any time. Latency is lower when you send a higher volume of events.
+- Batches may have to up 50 events, currently. This, too, is subject to change.
+- Batch sizes are not guaranteed. Due to the way that batches are accumulated internally, you may see smaller batch sizes than you expect when sending low rates of events.
+
+Additionally, you’ll need to coordinate with Segment’s R&D team for the time being. Please reach out to us in your dedicated Slack channel!
+
+## Quick Setup Actions
```
Comment (dlasky) [inline README.md:376]: quick setup is 'gone' so maybe we should call this section 'Default Subscriptions' ?

**Review by dlasky — APPROVED** [review]

**Pull request merged**